Source/EasyNetQ/Logging/LoggerExtensions.cs

```diff
@@ -0,0 +1,460 @@
+        }
+
+        public static void TraceException(this ILogger logger, string message, Exception exception, object formatParam0,
+            object formatParam1)
+        {
+            if (logger.IsTraceEnabled())
+                logger.Log(LogLevel.Trace, message.AsFunc(), exception, formatParam0, formatParam1);
+        }
+
+        public static void TraceException(this ILogger logger, string message, Exception exception, object formatParam0,
+            object formatParam1, object formatParam2)
+        {
+            if (logger.IsTraceEnabled())
+                logger.Log(LogLevel.Trace, message.AsFunc(), exception, formatParam0, formatParam1, formatParam2);
+        }
+
+        public static void Warn(this ILogger logger, Func<string> messageFunc)
+        {
+            GuardAgainstNullLogger(logger);
+            logger.Log(LogLevel.Warn, WrapLogInternal(messageFunc), null, EmptyParams);
+        }
+
+        public static void Warn(this ILogger logger, string message)
+        {
+            if (logger.IsWarnEnabled()) logger.Log(LogLevel.Warn, message.AsFunc(), null, EmptyParams);
+        }
+
+        public static void Warn(this ILogger logger, string message, params object[] args)
+        {
+            logger.WarnFormat(message, args);
+        }
+
+        public static void Warn(this ILogger logger, Exception exception, string message, params object[] args)
+        {
+            logger.WarnException(message, exception, args);
+        }
+
+        public static void WarnFormat(this ILogger logger, string message, params object[] args)
+        {
+            if (logger.IsWarnEnabled()) logger.LogFormat(LogLevel.Warn, message, args);
+        }
+
+        public static void WarnException(this ILogger logger, string message, Exception exception,
+            params object[] formatParams)
+        {
+            if (logger.IsWarnEnabled()) logger.Log(LogLevel.Warn, message.AsFunc(), exception, formatParams);
+        }
+
+        // ReSharper disable once UnusedParameter.Local
+        private static void GuardAgainstNullLogger(ILogger logger)
+        {
+            if (logger == null) throw new ArgumentNullException(nameof(logger));
+        }
+
+        private static void LogFormat(this ILogger logger, LogLevel logLevel, string message, params object[] args)
+        {
+            logger.Log(logLevel, message.AsFunc(), null, args);
+        }
+
+        // Avoid the closure allocation, see https://gist.github.com/AArnott/d285feef75c18f6ecd2b
+        private static Func<T> AsFunc<T>(this T value) where T : class
+        {
+            return value.Return;
+        }
+
+        private static T Return<T>(this T value)
+        {
+            return value;
+        }
+
+        // Allow passing callsite-logger-type to LogProviderBase using messageFunc
+        internal static Func<string> WrapLogSafeInternal(LoggerExecutionWrapper logger, Func<string> messageFunc)
```

Comment:
Is it used somewhere?

Reply:
It's just there because I've copied the preexisting one